app/policies/application_policy.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# Base policy class
+class ApplicationPolicy
+  attr_reader :user, :record
```

Comment:
An `admin` should have permission to see and do any actions.
The default result for _action query methods_ (`index? create?` etc) should all call this method:

```
def allowed_default_access?
   user.admin?
end
```

So the action query methods (`index? create?` etc)  should

Reply:
> The default result for action query methods (index? create? etc) should all call this method:

Any authenticated user can create resources. Except `UserController#index` which only admins can access, all other controllers' index actions are accessible by authenticated users — what resources they get to access is defined by the policy scope.

I have used the `user.admin?` predicate as the default for `update` and `destroy` actions as you suggested.